### ISSUE TITLE
[🐛 Bug]:  after upgrading to 4.1.4, jenkins pipeline hanging indefinitely, waiting for parallel test completion and no visible error anywhere.

### ISSUE THREAD

**Issue by jaycapps** [SeleniumHQ/docker-selenium#1564]
### What happened?

after updating to 4.1.4, Jenkins CICD executes in parallel just fine but then hangs.  I can't find any errors in the hub logs or node logs, no indication that anything is wrong... Jenkins thinks it's still running and all containers appear to be healthy. 

NOTE: 4.1.3 works fine.  everything up till 4.1.4  Not sure what changed but the impetus for 4.1.4 was born of closing an error thrown by the event bus during Nessus scans.  As soon as that issue was addressed by 4.1.4, this new issue started happening. 
currently stuck using 4.1.3 and having to restart that after every vulnerability scan performed on our network. 

### Command used to start Selenium Grid with Docker

```shell
docker stack deploy -c <composer file path> grid
```

### Relevant log output

```shell
I have nothing to give you here because there is no error showing up in the logs... it's just behavior resulting in hung parallel tests in jenkins running selenium for browser testing of a LAMP stack. 
I see nothing in the FINE logs with tracing enabled. nowhere is there any info to indicate a problem.
```

### Operating System

Centos 7 : latest kernel

### Docker Selenium version (tag)

4.1.4-20220407

**Comment by diemol** [comment]
The fix for the Nessus thing was possible because someone provided a test to replicate the issue. Hopefully there is a way to provide a test for this too and then we can have a look.

**Comment by jaycapps** [comment]
makes sense and I'm not sure how to tell you to reproduce the issue but I have mad a couple of observations over the past couple of days. 
1. this issue does not occur if any test failures do NOT occur.  the grid passes
2. while using 4.1.3, we've been seeing frequent and seemingly random `Unable to execute request for an existing session: java.util.concurrent.TimeoutException` issues.   these occur frequently but with 4.1.3, jenkins was aware and then issued a retry, which usually resulted in a pass. After upgrading to 4.1.4, my guess is... we are still encountering the issue but instead of moving onto a retry, the jenkins agent just sits in waiting until we stop the job.

**Comment by jaycapps** [comment]
Think I've got a result.   the issue for us in swarm mode seems helped by the fact that NODE_SESSION_TIMEOUT is now working again.  after switching to 4.1.3 from grid 3... I had completely overlooked this setting because it wasn't working at one point.  after re-introducing this... no more random freezing unless that setting is not dialed in correctly.

**Comment by diemol** [comment]
That is correct, session timeout was not working properly and a couple of reported issues showed that. Basically the timeout was not being respected when the session was being inactive due to flying HTTP request.

Thanks for reporting back! I will close this issue then.
